### ISSUE TITLE
[Suggestion] Allow the viewer to set key commands

### ISSUE THREAD

**Issue by JunmoreHeavyBox** [leezer3/OpenBVE#966]
### Description

The default numeric keypad camera control in the viewer's is not very convenient for users without a numeric keypad.
I want to change the keys myself, but the viewers does not provide control configuration.

#
Leaving aside WASD and directional keys.

**Comment by JunmoreHeavyBox** [comment]
Haven't looked at this issue for a long time.

**Comment by JunmoreHeavyBox** [comment]
After releasing the settings button, the camera did not stop moving.
The button icons in the interface have not been updated.
The 'backward' key seems to be set to the value of 'forward', and
[optionSaveErrorLog.txt](https://github.com/user-attachments/files/17114653/optionSaveErrorLog.txt)

**Comment by leezer3** [comment]
My fault- forgot to handle the key-up events. I've fixed this.

I'm not sure why closing the options dialog is failing for you- It doesn't for me. What did you do before you attempted to close the dialog?

**Comment by JunmoreHeavyBox** [comment]
I manually entered instead of using the dropdown menu to change keys. so

**Comment by leezer3** [comment]
OK, that might not work right- Not expected you to try that :)

If you use the dropdown, this should be OK. I'll disable typing in there for the next build (might be a few days)

**Comment by JunmoreHeavyBox** [comment]
I mention the remaining camera reset commands and camera controls related to the route viewer.
For the latter, more camera controls may have some key conflicts.

**Comment by JunmoreHeavyBox** [comment]
Urge...

![Problem of key definition](https://github.com/user-attachments/assets/8ba6cb66-26e4-4654-ab92-15cdd5ff69c3)

And

> OK, that might not work right- Not expected you to try that :)
> 
> If you use the dropdown, this should be OK. I'll disable typing in there for the next build (might be a few days)

**Comment by leezer3** [comment]
I can't reproduce the issue in your image (although this may be because of the re-write of the CFG loader in relatively recent builds)
Note that the CFG is only saved when the window is closed however.

I've fixed the editing of the boxes though, thanks for reminding me.

**Comment by JunmoreHeavyBox** [comment]
Thank you.

**Comment by JunmoreHeavyBox** [comment]
I want to explain.

When I confirmed the title, I think I used the plural form of Viewers, and I think I should mention both Object Viewer and RouteViewer. But this also brings some descriptive issues. I apologize.

I apologize for not providing an explanation for the unfinished task as I was eager to close the issue.

Probably
- Allow the RouteViewer to set key commands.
- The button icons in the interface have not been updated due to key changes.
